src/Database/QueryCompiler.php

```diff
@@ -211,9 +212,15 @@ protected function _buildJoinPart($parts, $query, $generator)
     {
         $joins = '';
         foreach ($parts as $join) {
+            $subquery = $join['table'] instanceof Query || $join['table'] instanceof QueryExpression;
             if ($join['table'] instanceof ExpressionInterface) {
-                $join['table'] = '(' . $join['table']->sql($generator) . ')';
+                $join['table'] = $join['table']->sql($generator);
```

Comment:
That's added added here: https://github.com/cakephp/cakephp/pull/8347/files#diff-e71e42d8a65ac8e52edc9657223c9f8cR222

The reason why we can't add the parentheses here is because the `CrossSchemaTableExpression` would be wrapped with parentheses which it shouldn't.